src/WebJobs.Extensions.DurableTask.CodeGeneration/WebJobs.Extensions.DurableTask.CodeGen.SourceGenerator/Utils/SyntaxNodeUtility.cs

```diff
@@ -0,0 +1,295 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using WebJobs.Extensions.DurableTask.CodeGeneration.SourceGenerator.Models;
+
+namespace WebJobs.Extensions.DurableTask.CodeGeneration.SourceGenerator.Utils
+{
+    public static class SyntaxNodeUtility
```

Comment:
We've got a lot of similar helper functions in our existing Durable Analyzer. It makes me wonder if we could reduce some duplicate code, but that could be saved for a future PR.

Reply:
Yes, there is a lot of overlap when parsing the functions.